backend/migration/utils.py

```diff
@@ -0,0 +1,131 @@
+from typing import Dict
+
+from qdrant_client._pydantic_compat import to_dict
+from qdrant_client.client_base import QdrantBase
+from qdrant_client.http import models
+
+from backend.logger import logger
+
+
+def migrate(
+    source_client: QdrantBase,
+    dest_client: QdrantBase,
+    source_collection_name: str,
+    destination_collection_name: str,
+    batch_size: int = 100,
+    same_qdrant: bool = False,
+) -> None:
+    """
+    Migrate collections from source client to destination client
+
+    Args:
+        source_client (QdrantBase): Source client
+        dest_client (QdrantBase): Destination client
+        source_collection_name (str): source collection names to migrate.
+        destination_collection_name (str): destination collection name.
+        recreate_on_collision (bool, optional): If True - recreate collection if it exists, otherwise
+            raise ValueError.
+        batch_size (int, optional): Batch size for scrolling and uploading vectors. Defaults to 100.
+    """
+    if _has_custom_shards(source_client, source_collection_name):
+        raise ValueError("Migration of collections with custom shards is not supported yet")
+    
+    if source_collection_name == destination_collection_name and same_qdrant:
+        destination_collection_name = destination_collection_name + "copy"
+        logger.debug(f"Destination collection name is same as source collection name. Renaming destination collection to {destination_collection_name}")
+
+    _recreate_collection(
```

Comment:
What if destination qdrant is different but destination collection exists? we should either break or just accept the collection as it instead of trying to re-create it

Reply:
ok, we arent checking if the destination qdrant exists or not yet. we might need to do that before migration. How about setting an override flag? If that flag is set, and destination collection exists, we recreate else we break?